compiler/rustc_hir_typeck/src/autoderef.rs

```diff
@@ -59,6 +59,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                                     region,
                                     mutbl,
                                     span: autoderef.overloaded_span(),
+                                    context: self.body_id.owner.to_def_id(),
```

Comment:
using `HirId` owner is wrong here :thinking: should be `tcx.hir().local_def_id(..)` iirc. Should become irrelevant with #103902